### PR TITLE
Introduce man pages and command to generate them

### PR DESCRIPTION
**Pull request by fraggerfox**
The PR adds support for generating UNIX style man pages for section 7, by utilizing the existing documentation for the commands.

```
$ jira man generate -r -o ./man-pages
```

Will generate the man pages for all the available commands in the utility.

It also adds the generated list of man pages to `/man` folder for easy pick up and install for package managers which are interested.

### CODE REVIEW

**Review by ankitpokhrel — COMMENTED** [review]
Hi @fraggerfox, Thank you so much for the PR. I've some comments.

- I think the `generate` sub-command is redundant. We can use a flag instead.
    ```
    $ jira man --generate
    ```
- Can we move generated man-pages inside `docs/man-pages` please?

Cheers,
Ankit

File: go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/briandowns/spinner v1.16.0
 	github.com/charmbracelet/glamour v0.3.0
 	github.com/cli/safeexec v1.0.0
+	github.com/cpuguy83/go-md2man/v2 v2.0.1 // indirect
```
[the remark this answers is not in the document]
Comment (fraggerfox) [reply go.mod:12]: This is the dependency added by `cobra/doc`.

https://github.com/spf13/cobra/blob/eb3b6397b1b5d1b0a2cd66a9afe0520f480c0a87/doc/man_docs.go#L27

File: internal/cmd/man/generate/generate.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+	if err != nil {
+		cmdutil.ExitIfError(err)
+	}
+}
+
+// SetFlags sets flags supported by a generate command.
+func SetFlags(cmd *cobra.Command) {
+	cmd.Flags().SortFlags = false
+
+	cmd.Flags().StringP("output", "o", "", "Name of the directory where the man pages would be generated")
```
[the remark this answers is not in the document]
Comment (fraggerfox) [reply internal/cmd/man/generate/generate.go:72]: Yeah, this is a better suggestion, it is safe to assume `/tmp` has write access by the user.

File: internal/cmd/man/generate/generate.go [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+	cmd.Flags().BoolP("create", "r", false, "Create the directory if it does not exist")
```
[the remark this answers is not in the document]
Comment (fraggerfox) [reply internal/cmd/man/generate/generate.go:73]: Yes, that is an option, I was just emulating the behavior of `-p` for `mkdir`. I can change that logic.

File: internal/cmd/man/generate/generate.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	createManLocation, err := cmd.Flags().GetBool("create")
+	cmdutil.ExitIfError(err)
+
+	manLocationName, err := cmd.Flags().GetString("output")
+	cmdutil.ExitIfError(err)
+
+	// Point to the current directory if no directory name is passed in.
+	if manLocationName == "" {
+		manLocationName = "."
+	}
```
[the remark this answers is not in the document]
Comment (fraggerfox) [reply internal/cmd/man/generate/generate.go]: Agreed.

**Comment by fraggerfox** [comment]
> 
> 
> Hi @fraggerfox, Thank you so much for the PR. I've some comments.
> 
>     * I think the `generate` sub-command is redundant. We can use a flag instead.
>       ```
>       $ jira man --generate
>       ```
> 
>     * Can we move generated man-pages inside `docs/man-pages` please?
> 
> 
> Cheers,
> Ankit

This should be done.

File: internal/cmd/man/generate/generate.go [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+	// Check if the directory exists, based on the flag create or print an error.
+	_, err = os.Stat(manLocationName)
+	if os.IsNotExist(err) {
+		if createManLocation {
+			err = os.MkdirAll(manLocationName, os.ModePerm)
+			if err != nil {
+				cmdutil.ExitIfError(err)
```
[the remark this answers is not in the document]
Comment (fraggerfox) [reply internal/cmd/man/generate/generate.go]: Since the single file refactor, this is no longer needed.

**Review by ankitpokhrel — APPROVED** [review]

**Pull request merged**